be/src/olap/txn_manager.cpp

```diff
@@ -500,4 +514,32 @@ bool TxnManager::get_expire_txns(TTabletId tablet_id, SchemaHash schema_hash, Ta
     return true;
 }
 
+void TxnManager::get_partition_ids(const TTransactionId transaction_id, std::vector<TPartitionId>* partition_ids) {
+    ReadLock txn_rdlock(&_txn_map_lock);
+    auto it = _txn_partition_map.find(transaction_id);
+    if (it != _txn_partition_map.end()) {
+        for(int64_t partition_id : it->second) {
+            partition_ids->push_back(partition_id);
+        }
+    }
+}
+
+void TxnManager::_insert_txn_partition_map(int64_t transaction_id, int64_t partition_id) {
```

Comment:
_insert_txn_partition_map_unlock may be better, can indicate there is no lock in it.

Reply:
done